src/backend/cuda/blas.cpp

```diff
@@ -179,21 +287,25 @@ void gemm(Array<T> &out, af_mat_prop optLhs, af_mat_prop optRhs,
     if (oDims.ndims() <= 2) {
         if (rDims[bColDim] == 1) {
             dim_t incr = (optRhs == AF_MAT_NONE) ? rStrides[0] : rStrides[1];
+            if (is_same<T, half>::value) {
+                AF_ERROR(
+                    "GEMV does not support half, Please create an issue on "
```

Comment:
is it impossible to gemv f16 arrays or just to be implemented ?

Reply:
It will have to be implemented internally. Its not something that is part of cuBLAS